src/commands/install/utils/extract_info_from_repo_id.py

```diff
@@ -0,0 +1,90 @@
+import re
+from typing import NamedTuple, Optional
+
+from .. import installation_exceptions
+
+
+class ExtractInfoFromRepoIdResult(NamedTuple):
+    package_name: str
+    tag: Optional[str]
+    url: Optional[str]
+
+
+def extract_info_from_repo_id(repo_id: str) -> ExtractInfoFromRepoIdResult:
+    result: Optional[ExtractInfoFromRepoIdResult] = None
+
+    if repo_id.startswith("git@"):
+        slug = _extract_slug_from_ssh(repo_id)
+        splitted_account_repo_name = slug.split("/")
+        if len(splitted_account_repo_name) == 2:
+            result = ExtractInfoFromRepoIdResult(
+                package_name=splitted_account_repo_name[1],
+                tag=None,
+                url=_map_ssh_to_url(repo_id),
+            )
+    elif ".org" in repo_id or ".com" in repo_id:
+        slug = _extract_slug_from_url(repo_id)
+        splitted_account_repo_name = slug.split("/")
+        if len(splitted_account_repo_name) == 2:
+            result = ExtractInfoFromRepoIdResult(
+                package_name=splitted_account_repo_name[1], tag=None, url=repo_id
+            )
+    else:
+        splitted_account_repo_name = repo_id.split("/")
+
+        if len(splitted_account_repo_name) == 2:
+            repo_name_with_tag = splitted_account_repo_name[1]
+            splitted_repo_name_tag = repo_name_with_tag.split("@")
+            tag: Optional[str] = None
+            if len(splitted_repo_name_tag) == 2:
+                tag = splitted_repo_name_tag[1]
+
+            result = ExtractInfoFromRepoIdResult(
+                package_name=splitted_repo_name_tag[0],
+                tag=tag,
+                url=_map_name_to_url(
+                    splitted_account_repo_name[0] + "/" + splitted_repo_name_tag[0]
+                ),
+            )
+
+    if result is None:
+        raise installation_exceptions.InvalidPackageName()
+
+    (package_name, tag, url) = result
+    return ExtractInfoFromRepoIdResult(
```

Comment:
with dataclass, you can use `dataclasses.replace(...)`  here for simpler prop substitutions